eigrpd/eigrp_interface.c

```diff
@@ -319,7 +319,7 @@ eigrp_if_up (struct eigrp_interface *ei)
   ne->ei = ei;
   ne->reported_metric = metric;
   ne->total_metric = metric;
-  ne->distance = eigrp_calculate_metrics (eigrp, &metric);
+  ne->distance = eigrp_calculate_metrics (eigrp, metric);
```

Comment:
passing the `struct eigrp_metrics` is a sure fire way to kill performance.  We need to figure out how to pass the data structure.